documentation/projects/proposals/project_improvement/20230914-implementation_plan_project_automations.md

```diff
@@ -0,0 +1,262 @@
+     - Rename "Merged" to "✅ Done" for consistency with issues board.
+
+### Improvements to existing automations
+
+1. [Issues/Issue is created](/meta/project_boards/issues.md#issue-is-created)
+   and
+   [Issues/Issue is added to the project](/meta/project_boards/issues.md#issue-is-added-to-the-project)
+
+   The issue should be added to the project board under the appropriate column,
+   "📋 Backlog" or "⛔ Blocked" depending on the issue labels.
+
+2. [Issues/Issue is closed](/meta/project_boards/issues.md#issue-is-closed)
+
+   The issue should be moved to the "✅ Done" column only if it was resolved. If
+   it was closed as discarded, it should be moved to a new "🗑️ Discarded"
+   column.
+
+   The built-in workflow will have to be replaced by a custom one that looks at
+   the `state_reason` field inside the issue payload of the `closed` activity of
+   the
+   [`issues` event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issues).
+
+3. [Issues/Issue is closed and inactive](/meta/project_boards/issues.md#issue-is-closed-and-inactive)
+
+   The current system means that issues are archived one by one as they become
+   inactive and with an 8 day cadence, this will slowly drift out of sync with
+   the weekly development chat.
+
+   Archiving should be performed by a custom scheduled workflow that archives
+   all items in the "✅ Done" and "🗑️ Discarded" columns weekly (preferably
+   immediately after the chat).
+
+4. [PRs/PR is created](/meta/project_boards/prs.md#pr-is-created) and
+   [PRs/PR is added to project](/meta/project_boards/prs.md#pr-is-added-to-the-project)
+
+   The PR should be added to the project board under the appropriate column, "In
+   Progress" or "Needs &lt;x&gt; Review(s)" depending on the PR's draft or
+   ready-for-review state and the number of existing reviews.
+
+5. [PRs/PR is closed or merged](/meta/project_boards/prs.md#pr-is-closed-or-merged)
+
+   The PR should be moved to the "✅ Done" column only if it was merged. If it
+   was closed without merge, it should be moved to a new "🗑️ Discarded" column.
+
+   This can be implemented by updating built-in workflows.
+
+### Restoration of previous automations
+
+1. Issues/Issue is connected to PR
+
+   When an issue is
+   [connected](https://docs.github.com/en/rest/overview/issue-event-types?apiVersion=2022-11-28#connected)
+   to a PR, the issue should move from "📋 Backlog" or "📅 To do" to the "🏗 In
+   Progress" column.
+
+   Unfortunately, the `connected` event does not trigger workflows. So this can
+   be implemented using a combination of multiple workflows.
+
+   - scheduled workflow
+   - [`pull_request` event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request)
+     - created
+     - edited
+
+2. Issues/PR linked to issue is closed without merge
+
+   When the PR for an issue in "🏗️ In Progress" is closed without merge, the
+   issue should go back to the "📅 To do" column.
```

Comment:
I think in these cases, it might be prudent to have it go to the backlog instead 🤔 We usually close PRs because we haven't had time to address the PR, or a community member hasn't responded to a PR they opened. In both of those cases, moving the issue to the backlog seems the most reasonable to me. What do you think?

Reply:
Moving to the Backlog does make sense 👍. As a possibility, we can also move it back to the column it came from.